android/src/main/java/com/wultra/android/powerauth/reactnative/PowerAuthRNModule.java

```diff
@@ -83,47 +84,120 @@ public void isConfigured(@Nonnull String instanceId, final Promise promise) {
+        final String masterServerPublicKey = map.getString("masterServerPublicKey");
+        if (baseEndpointUrl == null || appKey == null || appSecret == null || masterServerPublicKey == null) {
+            return null;
+        }
+        return new PowerAuthConfiguration.Builder(
                 instanceId,
                 baseEndpointUrl,
                 appKey,
                 appSecret,
                 masterServerPublicKey
         ).build();
+    }
 
-        PowerAuthClientConfiguration.Builder paClientConfigBuilder = new PowerAuthClientConfiguration.Builder();
-
+    /**
+     * Convert ReadableMap to {@link PowerAuthClientConfiguration} object.
+     * @param map Map with client configuration.
+     * @return {@link PowerAuthClientConfiguration} created from given map.
+     */
+    private static @NonNull PowerAuthClientConfiguration getPowerAuthClientConfigurationFromMap(final ReadableMap map) {
+        final boolean enableUnsecureTraffic = map.hasKey("enableUnsecureTraffic") && map.getBoolean("enableUnsecureTraffic");
+        final int connectionTimeout = map.hasKey("connectionTimeout") ? map.getInt("connectionTimeout") * 1000 : PowerAuthClientConfiguration.DEFAULT_CONNECTION_TIMEOUT;
+        final int readTimeout = map.hasKey("readTimeout") ? map.getInt("readTimeout") * 1000 : PowerAuthClientConfiguration.DEFAULT_READ_TIMEOUT;
+        final PowerAuthClientConfiguration.Builder paClientConfigBuilder = new PowerAuthClientConfiguration.Builder();
         if (enableUnsecureTraffic) {
             paClientConfigBuilder.clientValidationStrategy(new HttpClientSslNoValidationStrategy());
             paClientConfigBuilder.allowUnsecuredConnection(true);
         }
-        try {
-            configure(instanceId, new PowerAuthSDK.Builder(paConfig).clientConfiguration(paClientConfigBuilder.build()));
-            promise.resolve(true);
-        } catch (Exception e) {
-            promise.reject(EC_REACT_NATIVE_ERROR, "Failed to configure");
-        }
+        paClientConfigBuilder.timeouts(connectionTimeout, readTimeout);
+        return paClientConfigBuilder.build();
     }
 
-    @ReactMethod
-    public void deconfigure(String instanceId, final Promise promise) {
-        this.instances.remove(instanceId);
-        promise.resolve(null);
+    /**
+     * Convert ReadableMaps to {@link PowerAuthKeychainConfiguration} object.
+     * @param keychainMap Map with keychain configuration.
+     * @param biometryMap Map with biometry configuration.
+     * @return {@link PowerAuthKeychainConfiguration} created from given maps.
+     */
+    private static @NonNull PowerAuthKeychainConfiguration getPowerAuthKeychainConfigurationFromMap(final ReadableMap keychainMap, final ReadableMap biometryMap) {
+        // Biometry configuration
+        final boolean linkItemsToCurrentSet = biometryMap.hasKey("linkItemsToCurrentSet") && biometryMap.getBoolean("linkItemsToCurrentSet");
```

Comment:
This whole `hasKey &&` could lead to misconfiguration. For example, the `linkItemsToCurrentSet` default for Android is `true` (default in TS/JS), but if the Readable map will be missing such property (and it can when the user will create the object from scratch without this property), it will defaults to `false` because of the `hasKey` check.

Reply:
Oh, I see what you mean, I had to solve a similar problem for connection timeouts. I'll fix it.